test/scenarios/release/release_test.go

```diff
@@ -30,8 +35,13 @@ func TestMain(m *testing.M) {
 }
 
 func TestRelease(t *testing.T) {
+	usesOldVersion := false
+	usedSemVer, _ := dtversion.ToSemver(releaseTag)
+	if semver.Compare(thresholdVersion, usedSemVer) >= 1 {
+		usesOldVersion = true
+	}
```

Comment:
we are comparing basically 2 `const` here, where the outcome is always the same, why are we doing this? 😅 

is there a case where the "upgrade" test scenario will become a "downgrade" test scenario? (I hope not 😅 )

can we determine the `usesOldVersion` is any nice way?
- meaning: will there be some check that will magically tell us if version A supports the new CR version that version B uses?
  - I don't think so, or at least not now

Reply:
My thought process was: currently we are at `1.1.0`, which does not know `v1beta2`. When `1.2.0` is released we have a version that knows `v1beta2`. From then on (after we bump the `releaseTag` in the test manually, because it is hardcoded) we always use the normal install method. After that the here introduced changes become useless and may be removed. But at the moment I do not see a other way to fix this test. What do you think about this? Do you have some tips what I could change here?

Reply:
There are multiple perspectives:
1. **Does it hurt to always check if the conversion works properly?** (ie. just always consider `usesOldVersion` as `true`)
  - IMO, it should be fine to check if the conversion is working properly
    - kinda mixes the responsibilities of the tests, which is not nice, but not horrible 
  - this results in a "dumber" logic when it comes to the e2e test code (we don't have extra checks)
2. **Automate it to the fullest:**
  - So we never have to manually update any values, it "just works" and tests the upgrade from `latest released version` to the `current branch`.
  - Your check would kinda make sense in that situation
  - For this to work the following would need to be true:
     1. We (dynamically) know what is the `latest released version`
     2. We (dynamically) know what is the `latest CR version` that the `latest released version` supports
     3. We (dynamically) know what is the `latest CR version` that the `current branch` supports
     4. We (dynamically) know what is the "oldVersion" to use to convert to 🙃 
     5. ...
3. **Partially automate it**
 - So we only have to intervene, when a new version is introduced 
 - This is what you did
   - More accurately, "would have done", if the `git` magic worked 😅 


I do not think option 2. is anyway feasible.

I like option 1, for its simplicity, but can be annoying to keep up to date

Option 3 would make sense and would be the best "middle ground" IF the `git` magic can be used.
- As of right now we do not now
- Introducing code that "might be used in the future" is something we should avoid, as it will only cause confusion.
  - If I asked anyone, 2 weeks from now, "Why do we have that code?", the likely answer would be "dunno, does not make much sense, but don't want to touch it as it works, right?"

Reply:
I guess the most practical way currently is to go for option 1, then. I will go for it. Thank you for elaborating.

Reply:
np, as mentioned before, we should look into **option 3.** separately  😉

Reply:
done in https://github.com/Dynatrace/dynatrace-operator/pull/3205/commits/ba567e6920800e3d0d4ab60466bb2464fe1e8390

I removed the whole logic with the check if we use an old version because if we assume it is always true we do not need that flag.